math/mathcore/test/testTMath.cxx

```diff
@@ -181,6 +182,46 @@ void testBreitWignerRelativistic()
   }
 }
 
+void testHalfSampleMode()
+{
+   const long testdata_n = 50;
+   double testdata[testdata_n] =
+     {-1.8626292050574662, -1.2588261580948075, -1.2148747383283962, -0.88052174765194313,
+      -0.85819166488158083, -0.70330273835955692,-0.62581680819121988, -0.60199237302865982,
+      -0.056104207433014919, -0.048469532592846587, -0.045160899289979461, 0.00090904300338501276,
+      0.07801333189364168, 0.13270976192391337, 0.16400228933346139, 0.20005259782423812,
+      0.20717036575551762, 0.2161088542792593, 0.24097449144042737, 0.24416261486911417,
+      0.36874321835120116, 0.37189144575412991, 0.3890935935298675, 0.40520928475289097,
+      0.42184335741753576, 0.45308183033935723, 0.47769262360841214, 0.48905822724024939,
+      0.48979853918045224, 0.53563861071255214, 0.61398403826022885, 0.62855905995409977,
+      0.63978982255456396, 0.67136185490980282, 0.68026440980784897, 0.7130887489094474,
+      0.71656371678168229, 0.72363941257687625, 0.75056685823912761, 0.75558971232198124,
+      0.79884984432253403, 0.8188843130575223, 0.83576827230628292, 0.84235396751558722,
+      0.85809047760873358, 0.89413700981004551, 0.9040856696277757, 0.91910499099412069,
+      0.92055153154640645, 0.9373728168229567};
+   double testw[testdata_n] {}; // all equal zero weights
+   
+   //const long testdata2_n = 16;
+   //unsigned short testdata2[testdata2_n] = {0,0,0,0,0,
+   //                                 1,1,1,1,1,
+   //                                 2,2,2,2,2,2};
+   
+   const long testdata3_n = 4;
+   double testdata3[testdata3_n] = {1,2,3,3.25};
+   
+   //const long testdata4_n = 10;
+   //unsigned short testdata4[testdata4_n] = {0,0,0,1,1,1,1,2,2,2};
```

Comment:
This test seems to be particularly important, since it's the only one with duplicate numbers. Could you enable it, and also "unsort" the input data?

Further, there is no test with weights, so we should also have that probably.